crates/proto/src/serialize/txt/trust_anchor.rs

```diff
@@ -0,0 +1,273 @@
+//! Trust anchor file parser
+//!
+//! A trust anchor file largely adheres to the syntax of a zone file but may only contain
+//! DNSKEY records
+
+use std::{borrow::Cow, str::FromStr as _};
+
+use crate::{
+    rr::{dnssec::rdata::DNSKEY, DNSClass, Name, Record, RecordType},
+    serialize::txt::ParseErrorKind,
+};
+
+use super::{
+    rdata_parsers::dnskey,
+    zone,
+    zone_lex::{Lexer, Token as LexToken},
+    ParseError, ParseResult,
+};
+
+/// Trust anchor file parser
+pub struct Parser<'a> {
+    lexer: Lexer<'a>,
+}
+
+impl<'a> Parser<'a> {
+    /// Returns a new trust anchor file parser
+    pub fn new(input: impl Into<Cow<'a, str>>) -> Self {
+        Self {
+            lexer: Lexer::new(input),
+        }
+    }
+
+    /// Parse a file from the Lexer
+    ///
+    /// Returns the DNSKEY records found in the file
+    pub fn parse(mut self) -> ParseResult<Vec<Record<DNSKEY>>> {
```

Comment:
we could future proof and return `Vec<Entry>` where the only variant `Entry` has today is `Record<DNSKEY>`. In the future, we'll add another variant for `Record<DS>`.

does hickory-proto strictly adhere to semver or is it considered an implementation detail library not to be directly used?

Reply:
We adhere to semver, but also most releases break semver.